pilot/pkg/serviceregistry/kube/controller/endpoint_builder.go

```diff
@@ -93,13 +97,14 @@ func NewEndpointBuilderFromMetadata(c controllerInterface, proxy *model.Proxy) *
 			Label:     locality,
 			ClusterID: c.Cluster(),
 		},
-		tlsMode: model.GetTLSModeFromEndpointLabels(proxy.Labels),
+		tlsMode:  model.GetTLSModeFromEndpointLabels(proxy.Labels),
+		nodeName: proxy.Metadata.NodeName,
 	}
 	var networkID network.ID
 	if len(proxy.IPAddresses) > 0 {
 		networkID = out.endpointNetwork(proxy.IPAddresses[0])
 	}
-	out.labels = labelutil.AugmentLabels(proxy.Labels, c.Cluster(), locality, networkID)
+	out.labels = labelutil.AugmentLabels(proxy.Labels, c.Cluster(), locality, proxy.Metadata.NodeName, networkID)
```

Comment:
we use GetNodeName in other places, is this intended?

Reply:
No, have changed to `proxy.GetNodeName()`